### DISCUSSION TITLE
Run Create3 via ROS 2 Create3_Bluetooth Node?

### DISCUSSION THREAD

**Discussion by slowrunner** [iRobotEducation/create3_docs#614] [Ideas]
I'm still feeling separation anxiety from my failed attempt to find a configuration to successfully run the Create3 with an Oak-D camera and RTABmap.  I found a very old youtube with a Pi4 powered TB4 successfully running RTABmap with an Oak-D-Pro camera, but I could not get my Galactic or Humble Create3 to stay alive with the added topic traffic, regardless of the RMW/DDS configuration and override parameters.

Some people have noted that the cyclone and fastdds RMW require memory allocation for all topics regardless of subscription which may have caused the Create3 resources to become overloaded.  The great iRobot devs attempted to help me segment and bridge the RMW/DDS, but we were unable to find a working solution. 

It came to me that one thing I did not try was coding a "ROS 2 Create3_Bluetooth" node that would run on the Pi4 or Pi5 (in Docker) to bridge the ROS topics to the Create3 using the Create3 Python Web Playground Bluetooth connection.  I don't know what sensor publishing latency and rate could be supported, or robot.set_wheel_speeds(,) latency from /cmd_vel could be supported, but this would totally offload all ROS 2 messaging from the Create3.  

I'm guessing from the discussions that this approach has not been tried yet, although I also have not seen an abundance of folks with overloaded Create3 either...

Just a thought.

**Comment by alsora** [comment]
Hi @slowrunner,

> iRobot devs attempted to help me segment and bridge the RMW/DDS, but we were unable to find a working solution.

The Create 3 has now been updated to Jazzy and we have released the package to bridge the DDS topics: https://github.com/iRobotEducation/create3_examples.
There are small improvements with respect to what you helped us prototype (and I really thank you for that!).
Most importantly, the Turtlebot team is going to release scripts to setup the republisher on a Raspberry Pi (they are going to include it in the next release of  the Turtlebot ROS 2 packages).
So maybe you may give it a try once it's available =)

> bridge the ROS topics to the Create3 using the Create3 Python Web Playground Bluetooth connection

Unfortunately I'm not really familiar with the Python Bluetooth support.
You could take a look at the Python SDK: https://github.com/iRobotEducation/irobot-edu-python-sdk
I don't know the performance of the approach and if all the data that you need are published there.

I will try to get some information and get back to you.

**Comment by slowrunner** [comment]
Thanks for the update.
